cosmos.md

```diff
@@ -0,0 +1,198 @@
+information, as described below. 
+
+Note: An empty or null `chain_id` must be treated as an error.
+
+### Direct References
+
+If the `chain_id` matches the case-sensitive pattern `[-a-zA-Z0-9]{1,32}` and
+does not start with "hashed-", it can be assumed to refer to a `chain_id`.  If
+the first 7 characters are `hashed-`, however, it should be assumed to be a hash
+of a `chain_id`; the original can be retrieved from any node as described below.  
+
+Note: some `chain_id`s include a version in the format `-version-X` where X is
+an integer. For more information, see [ICB#517][].
+
+### Hashed References
+
+References prefixed by `hashed-` are defined as
+`first_16_chars(hex(sha256(utf8(chain_id))))`, with:
+
+- `chain_id`= the Tendermint `chain_id` from the genesis file (see above)
+- `utf8`= a UTF-8 encoding function
+- `sha256`= a SHA256 hash function
+- `hex`= a lowercase hex encoder
+- `first_16_chars`= a function to extract the first 16 characters of the
+  resulting string and dropping the rest
+
+### Resolution Method
+
+To resolve a blockchain reference for the Cosmos namespace, make a REST GET
+request to the blockchain node with endpoint `/node_info`, for example:
+
+```jsonc
+
+//TODO -- RPC ENDPOINT IS OUT OF DATE
+
+// Request
+curl -X GET "https://stargate.cosmos.network/node_info" -H "accept: application/json"
+
+// Response
+{
+  "application_version": {
+    "build_tags": "string",
+    "client_name": "string",
+    "commit": "string",
+    "go": "string",
+    "name": "string",
+    "server_name": "string",
+    "version": "string"
+  },
+  "node_info": {
+    "id": "string",
+    "moniker": "validator-name",
+    "protocol_version": {
+      "p2p": 7,
+      "block": 10,
+      "app": 0
+    },
+    "network": "gaia-2",
+    "channels": "string",
+    "listen_addr": "192.168.56.1:26656",
+    "version": "0.15.0",
+    "other": {
+      "tx_index": "on",
+      "rpc_address": "tcp://0.0.0.0:26657"
+    }
+  }
+}
+```
+The response will return a JSON object which will include node information and
+the `chain_id` defined above can be retrieved from `node_info.network`; this can
+be used directly as the `reference` section of a CAIP-2 or CAIP-10 .
+
+## Rationale
+
+While there is no enforced restriction on `chain_id`, the authors of this
+document did not find a chain ID in the wild that does not conform to the
+restrictions of the direct reference definition above.
+
+Cosmos blockchains with a chain ID not matching `[-a-zA-Z0-9]{1,32}` or prefixed
+with "hashed-" would need to be hashed in order to conform to this namespace
+specification. No real world example of this case is known to the author yet.
+
+One unique trait of the Cosmos ecosystem is that if chainstate is broken, chains
+can be "rebooted" at a later time; when this happens, a new genesis block is
+generated from a new configuration object and new `chain_id`. This requires a
+versioning syntax for `chain_id`s using the `-version-X` suffix (where X is a
```

Comment:
The convention is just a number, not the word "version":

```suggestion
versioning syntax for `chain_id`s using the `-X` suffix (where X is a
```

I have never seen `-version-` in any chain ID.

Reply:
rejiggered